### PR TITLE
Add tochvision FCN Resnet50

### PR DESCRIPTION
**Pull request by oke-aditya**
Helps pytorch/vision#4289

Adds the following models.
- [x] FCN: - fcn_resnet50

Edit: -
I checked the netlify preview. It was as expected.
Feel free to merge.

cc @NicolasHug @datumbox

### CODE REVIEW

File: pytorch_vision_fcn_resnet101.md [hunk 1]
```diff
@@ -18,7 +18,9 @@ order: 10
 
 ```python
 import torch
-model = torch.hub.load('pytorch/vision:v0.10.0', 'fcn_resnet101', pretrained=True)
+model = torch.hub.load('pytorch/vision:v0.10.0', 'fcn_resnet50', pretrained=True)
```
Comment (NicolasHug) [inline pytorch_vision_fcn_resnet101.md:21]: was the change from 101 to 50 intended?

I would suggest to leave 101 because when running the example from the colab notebook, 101 ends up with a more accurate segmentation

File: pytorch_vision_fcn_resnet101.md [hunk 2]
```diff
@@ -35,7 +37,7 @@ So, `output['out']` is of shape `(N, 21, H, W)`. More documentation can be found
 ```python
 # Download an example image from the pytorch website
 import urllib
-url, filename = ("https://github.com/pytorch/hub/raw/master/images/dog.jpg", "dog.jpg")
+url, filename = ("https://github.com/pytorch/hub/raw/master/images/deeplab1.png", "deeplab1.png")
```
Comment (NicolasHug) [inline pytorch_vision_fcn_resnet101.md:40]: thanks for changing the image, it is indeed much better with this one

File: pytorch_vision_fcn_resnet101.md [hunk 3]
```diff
@@ -85,13 +88,14 @@ plt.imshow(r)
 
 ### Model Description
 
-FCN-ResNet101 is constructed by a Fully-Convolutional Network model with a ResNet-101 backbone.
+FCN-ResNet is constructed by a Fully-Convolutional Network model using respective ResNet-50 or ResNet-101 backbone.
```
Comment (NicolasHug) [inline pytorch_vision_fcn_resnet101.md:91]:
```suggestion
FCN-ResNet is constructed by a Fully-Convolutional Network model, using a ResNet-50 or a ResNet-101 backbone.
```

**Review by NicolasHug — APPROVED** [review]
Thanks a lot @oke-aditya , some minor comment / question but LGTM when addressed!

File: pytorch_vision_fcn_resnet101.md (hunk 1 above)
Comment (datumbox) [reply pytorch_vision_fcn_resnet101.md:21]: I think it's because of its popularity. ResNet50 ended up being used much more in applications and papers because it balances accuracy for speed. Given it's the most popular across the resnet v1 variants, it's not a bad idea to highlight this instead of the deeper version.

Comment (oke-aditya) [reply pytorch_vision_fcn_resnet101.md:21]: My idea was to download model with lower bandwidth. `resnet50` takes slightly less time in inference and has smaller model footprint. Thus speeding up CI slightly.

Comment (NicolasHug) [reply pytorch_vision_fcn_resnet101.md:21]: If it's just about CI, even 10 seconds for a single won't matter much. But the popularity of the variant makes sense. Thanks for the replies
